arrows/klv/klv_read_write.txx

```diff
@@ -343,6 +356,114 @@ klv_ber_oid_length( T value )
   return ( _int_bit_length( value ) + 6 ) / 7;
 }
 
+// ---------------------------------------------------------------------------
+template < class T, class Iterator >
+double
+klv_read_flint( double minimum, double maximum, Iterator& data, size_t length )
+{
+  KLV_ASSERT_INT( T );
+  KLV_ASSERT_UINT8_ITERATOR( data );
+
+  _check_range_length( minimum, maximum, length );
+
+  // Before read to avoid moving iterator
+  if( std::is_signed< T >::value && minimum != -maximum )
+  {
+    throw std::logic_error( "range must be symmetrical around zero" );
+  }
+
+  auto const int_value = klv_read_int< T >( data, length );
+  auto const float_value = static_cast< double >( int_value );
+
+  // C++17: if constexpr
+  if( std::is_signed< T >::value )
+  {
+    // Special invalid / out-of-range value
+    if( int_value == _int_min< T >( length ) )
+    {
+      return std::numeric_limits< double >::quiet_NaN();
+    }
+
+    auto const scale = maximum / _int_max< T >( length );
+    return float_value * scale;
+  }
+  else
+  {
+    auto const scale = ( maximum - minimum ) / _int_max< T >( length );
+    return float_value * scale + minimum;
+  }
+}
+
+// ---------------------------------------------------------------------------
+template < class T, class Iterator >
+void
+klv_write_flint( double value, double minimum, double maximum, Iterator& data,
+                 size_t length )
+{
+  // Ensure types are compatible with our assumptions
+  KLV_ASSERT_INT( T );
+  KLV_ASSERT_UINT8_ITERATOR( data );
+
+  _check_range_length( minimum, maximum, length );
+
+  // C++17: if constexpr
+  if( std::is_signed< T >::value )
+  {
+    if( minimum != -maximum )
+    {
+      throw std::logic_error( "range must be symmetrical around zero" );
+    }
+
+    auto const max_int = _int_max< T >( length );
+    auto const min_int = _int_min< T >( length );
+    auto const scale = max_int / maximum;
+    auto const float_value = value * scale;
+    auto int_value = static_cast< T >( std::round( float_value ) );
```

Comment:
I keep trying to decide whether I'm concerned that this will likely exhibit odd behavior if `length > 6`, due to the integer type having a greater range of distinct values than `double`. Thoughts?